streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -422,12 +446,14 @@ int numBuffered() {
     boolean maybePunctuate() {
         final long timestamp = partitionGroup.timestamp();
 
+        boolean punctuated = systemTimePunctuationQueue.mayPunctuate(System.currentTimeMillis(), PunctuationType.SYSTEM_TIME, this);
```

Comment:
we should not call `System.currentTimeMillis()` directly, but use the tasks `Time` object -- otherwise, we cannot test anything.